homeassistant/components/denonavr/__init__.py

```diff
@@ -60,12 +62,12 @@ async def async_setup_entry(
 
     # Connect to receiver
     connect_denonavr = ConnectDenonAVR(
-        hass,
         entry.data[CONF_HOST],
         DEFAULT_TIMEOUT,
         entry.options.get(CONF_SHOW_ALL_SOURCES, DEFAULT_SHOW_SOURCES),
         entry.options.get(CONF_ZONE2, DEFAULT_ZONE2),
         entry.options.get(CONF_ZONE3, DEFAULT_ZONE3),
+        get_async_client(hass),
     )
     if not await connect_denonavr.async_connect_receiver():
         raise ConfigEntryNotReady
```

Comment:
Please test that `async_connect_receiver` never takes longer than 10s when the receiver is turned off to ensure startup is not blocked if the receiver is unplugged.

Reply:
The timeout is 5 seconds. With 2 additional zones enabled it will be called 3 times. This is more than 10 seconds in total. But everthing is awaited and nothing blocking.
When I tested it with all zones enabled, I never got the 10 seconds warning message.